### PR TITLE
Added search functionality

### PR DESCRIPTION
**Pull request by dmulugisi**

### CODE REVIEW

File: src/containers/sidebar/index.js [hunk 1]
```diff
@@ -105,7 +110,7 @@ class ChatBar extends React.Component{
             <MuiThemeProvider>
                 <SelectableList defaultValue={0} changeChat={this._handleChatChange}>
                     <ListItem disabled={true}>
-                        <SearchBar hintText="Search Chats" />
+                        <TextField className="searchbar" hintText="Search Chats" value={this.setState.search} onChange={this._handleSearchChange.bind(this)} />
```
Comment (KimberleighBeyl) [inline src/containers/sidebar/index.js]: this.setState.search?

**Review by KimberleighBeyl — CHANGES_REQUESTED** [review]
why setstate

File: src/containers/sidebar/index.js (hunk 1 above)
Comment (dmulugisi) [reply src/containers/sidebar/index.js]: after searching, the chat-list has to be re-rendered again i used setState to ensure that something was typed in the searchbar then render the sorted chats list. Should I change it?

Comment (KimberleighBeyl) [reply src/containers/sidebar/index.js]: should be this.state.search

File: src/containers/sidebar/index.js [hunk 2]
```diff
@@ -66,17 +66,22 @@ class ChatBar extends React.Component{
...
+    _handleSearchChange(event) {
+        this.setState({ search: event.target.value.substr(0, 20) });
+        this.componentWillMount(this.props);
+    }
     componentWillMount(props){
         console.log("Mounting side bar")
         //console.log(this.props.chatList);
-        var chatItems = this.props.chatList;
+        let filteredChats = this.props.chatList.filter((chats) => { return chats.name.toLowerCase().indexOf(this.state.search.toLowerCase()) !== -1; });
+        var chatItems = filteredChats;
```
Comment (grootendorste) [inline src/containers/sidebar/index.js]: Why not just stick with let instead of using var?

File: src/containers/sidebar/index.js (hunk 1 above)
Comment (grootendorste) [reply src/containers/sidebar/index.js]: I agree. I think this.setState.search will be undefined.

File: src/containers/sidebar/index.js [hunk 3]
```diff
@@ -124,12 +129,12 @@ class NewChat extends React.Component{
             <MuiThemeProvider>
                 <List>
                 <ListItem disabled={true}>
-                    <FloatingActionButton mini={true} style={{marginLeft: 335}} backgroundColor="Grey" onClick={this.props.new_Chat}>
```
Comment (grootendorste) [inline src/containers/sidebar/index.js:127]: Why new_Chat and not newChat?

File: src/containers/sidebar/index.js [hunk 4]
```diff
@@ -124,12 +129,12 @@ class NewChat extends React.Component{
...
                 <List>
                 <ListItem disabled={true}>
-                    <FloatingActionButton mini={true} style={{marginLeft: 335}} backgroundColor="Grey" onClick={this.props.new_Chat}>
+                    <FloatingActionButton mini={true} style={{ "marginLeft": "85%" }} backgroundColor="Grey" onClick={this.props.new_Chat}>
                         <NavigationCancel />
                     </FloatingActionButton>
                     <br />
                     <br />
-                    <TextField hintText="Friends Email" style={{marginLeft: 55}} errorText="" onChange={this.showFriends}/>
+                    <TextField hintText="Friends Email" style={{ "marginLeft": "15%" }} errorText="" onChange={this.showFriends} />
```
Comment (grootendorste) [inline src/containers/sidebar/index.js:157]: Does this.showFriends exist?

**Review by grootendorste — CHANGES_REQUESTED** [review]

**Review by KimberleighBeyl — APPROVED** [review]

**Pull request merged**